components/search_hint/search_hint.tsx

```diff
@@ -1,32 +1,84 @@
+    options: SearchTerm[];
 }
 
-export default class SearchHint extends PureComponent<Props> {
-    public render() {
-        return (
-            <React.Fragment>
-                {this.props.withTitle &&
-                <h4>
+export const SearchHint = (props: Props) => {
+    const [index, setIndex] = useState(0);
+
+    const onKeyDown = (event: KeyboardEvent) => {
+        if (event.key === 'ArrowDown') {
+            const newIndex = index === props.options.length - 1 ? 0 : index + 1;
+            setIndex(newIndex);
+        }
+
+        if (event.key === 'ArrowUp') {
+            const newIndex = index === 0 ? props.options.length - 1 : index - 1;
+            setIndex(newIndex);
+        }
+
+        if (event.key === 'Enter') {
+            props.updateSearchTerms(props.options[index].searchTerm);
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener('keydown', onKeyDown);
+
+        return () => {
+            document.removeEventListener('keydown', onKeyDown);
+        };
+    });
+
+    return (
+        <React.Fragment>
+            {props.withTitle &&
+                <h4 className='search-hint__title'>
                     <FormattedMessage
                         id='search_bar.usage.title'
                         defaultMessage='Search Options'
                     />
                 </h4>
-                }
-                <FormattedMarkdownMessage
-                    id='search_bar.usage.tips'
-                    defaultMessage='* Use **"quotation marks"** to search for phrases\n* Use **from:** to find posts from specific users\n * Use **in:** to find posts in specific channels\n* Use **on:** to find posts on a specific date\n* Use **before:** to find posts before a specific date\n* Use **after:** to find posts after a specific date\n* Use **dash** "-" to exclude search terms and modifiers'
-                />
-            </React.Fragment>
-        );
-    }
-}
+            }
+            <ul
+                role='list'
+                className='search-hint__suggestions-list'
+                onMouseDown={props.onMouseDown}
+            >
+                {props.options.map((option, optionIndex) => (
+                    <li
+                        className={classNames('search-hint__suggestions-list__option', {highlighted: index === optionIndex})}
+                        key={option.searchTerm}
+                        onClick={() => props.updateSearchTerms(option.searchTerm)}
+                        onMouseOver={() => setIndex(optionIndex)}
+                    >
+                        <div className='search-hint__suggestion-list__flex-wrap'>
+                            <span className='search-hint__suggestion-list__label'>{option.searchTerm}</span>
+                        </div>
+                        <div className='search-hint__suggestion-list__value'>
+                            <FormattedMessage
+                                id={t(option.message.id)}
```

Comment:
`t` needs to receive a string literal for it to be picked up by the i18n tools, so this won't stop them from messing up `en.json`. You'll need to use `t` wherever `props.options` is defined.

Reply:
@nevyangelova This still needs to be looked at. This is why your search strings aren't being added to `en.json`, and I think it's why there's an entry being created for "undefined"`. You can wrap the IDs defined in the constants with `t` to fix this.